MIPS/mip-0006-slot-reduction-90s.md

```diff
@@ -0,0 +1,270 @@
+---
+mip: <number to be provided>
+title: Reduce slot time to 90s
+description: This MIP proposes to reduce the slot time in Mina's Ouroboros Samasika consensus protocol from 180 seconds to 90 seconds, enabling more frequent block creation and doubling network throughput while maintaining system stability.
+authors: George Agapov <george@o1labs.org>, Christian Despres <christian@o1labs.org>
+discussions-to: <to be provided> 
+status: Review
+type: Meta
+category: Engineering
+created: 2025-06-05
+---
+
+# Abstract
+
+Mina was the first succinct blockchain built entirely on zero-knowledge cryptography. We initially prioritized robust system functionality over performance to ensure Mina's reliability. With this protocol update, we're beginning a series of incremental performance enhancements that will bring Mina's finality and throughput in line with industry standards, while remaining fully committed to our original zero-knowledge vision.
+
+This MIP proposes to reduce the slot time in Mina's Ouroboros Samasika consensus protocol from the current 180 seconds (3 minutes) to 90 seconds (1.5 minutes). This change will allow for more frequent block creation while maintaining system stability and performance, made possible by recent improvements in block creation performance.
+
+We aim to deliver this MIP with a protocol upgrade scheduled for the Fall 2025.
+
+# Motivation
+
+Mina uses the Ouroboros Samasika consensus protocol, which divides time into equal-sized chunks called slots. Currently, the slot time is set at 3 minutes, which directly limits how frequently blocks in the Mina blockchain can be created, as consensus ensures at most one block per slot.
+
+The original slot time of 180 seconds was determined primarily by performance factors, including:
+
+1. The time required to create a Kimchi proof-system-based SNARK proof (5-15 seconds depending on hardware)
+2. The requirement that SNARK work for certain previous blocks must be completed before creating the current block
+3. Ensuring all nodes can receive blocks before the end of a slot
+4. Maximum number of account updates per zkapp transaction
+
+Recent protocol releases have dramatically improved block creation and transaction processing performance (see Appendix for details). These improvements now make it feasible to decrease the slot time without compromising system stability or performance.
+
+Reducing the slot time will allow for more frequent block creation, improving the network's throughput and user experience.
+
+# Specification
+
+Following protocol modifications are required to implement this change:
+
+1. Update the `block_window_duration_ms` constant from 180,000 to 90,000 milliseconds
+    - Effect: Slot time will be reduced from 3 minutes to 90 seconds
```

Comment:
Also consider/discuss effect on epoch length, including impact to payment operations for staking pools, etc.

Reply:
Added explanation to Security considerations